src/main/java/hudson/tasks/junit/TestResultSummary.java

```diff
@@ -21,6 +23,7 @@ public TestResultSummary(TestResult result) {
         this.skipCount = result.getSkipCount();
         this.passCount = result.getPassCount();
         this.totalCount = result.getTotalCount();
+        this.failedTests = result.getFailedTests();
```

Comment:
Maybe makes sense to double check that the collection is serializable, e.g. via copy constructor